core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpExecutionServiceDAO.scala

```diff
@@ -28,49 +29,47 @@ class HttpExecutionServiceDAO( executionServiceURL: String, submissionTimeout: F
   private implicit lazy val baseMetricBuilder =
     ExpandedMetricBuilder.expand(SubsystemMetricKey, Subsystems.Cromwell)
 
+  private def pipeline[A: FromResponseUnmarshaller](userInfo: UserInfo, retryCount: Int = 0) =
+    addAuthHeader(userInfo) ~> instrumentedGzSendReceive(retryCount) ~> unmarshal[A]
+
+  private def pipelineWithTimeout[A: FromResponseUnmarshaller](userInfo: UserInfo, timeout: Timeout, retryCount: Int = 0) =
+    addAuthHeader(userInfo) ~> instrumentedGzSendReceive(timeout, retryCount) ~> unmarshal[A]
+
   override def submitWorkflows(wdl: String, inputs: Seq[String], options: Option[String], userInfo: UserInfo): Future[Seq[Either[ExecutionServiceStatus, ExecutionServiceFailure]]] = {
     val timeout = Timeout(submissionTimeout)
     val url = executionServiceURL+"/workflows/v1/batch"
-
-    val pipeline = addAuthHeader(userInfo) ~> instrumentedGzSendReceive(timeout) ~> unmarshal[Seq[Either[ExecutionServiceStatus, ExecutionServiceFailure]]]
     val formData = Map("workflowSource" -> BodyPart(wdl), "workflowInputs" -> BodyPart(inputs.mkString("[", ",", "]"))) ++ options.map("workflowOptions" -> BodyPart(_))
-    pipeline(Post(url, MultipartFormData(formData)))
+    pipelineWithTimeout[Seq[Either[ExecutionServiceStatus, ExecutionServiceFailure]]](userInfo, timeout) apply (Post(url, MultipartFormData(formData)))
   }
 
   override def status(id: String, userInfo: UserInfo): Future[ExecutionServiceStatus] = {
     val url = executionServiceURL + s"/workflows/v1/${id}/status"
-    val pipeline = addAuthHeader(userInfo) ~> instrumentedGzSendReceive ~> unmarshal[ExecutionServiceStatus]
-    retry(when500) { () => pipeline(Get(url)) }
+    retry(when500) { count => pipeline[ExecutionServiceStatus](userInfo, count) apply Get(url) }
```

Comment:
This seems off to me. Why does the call being retried care how many times it's been retried? If it did, it would have implemented its own `recover`. Maybe I'm stuck in OO-land here but this seems to violate encapsulation or something. Not so much here but this is especially noticeable in the Google classes where we have to pass around the `implicit count =>` in every call we want to retry.

For the Google case -- what if there was a GoogleUtilities/Retry mashup mixin called `GoogleRetry` with its own Google-aware definitions of the retry functions? Could we then keep the retry count isolated in the retry functions, rather than having to pass them through everything retry-able so they can pop out in `executeGoogleCall`? I'm okay tracking retries as a separate metric to the HTTP requests themselves (i.e. not appending `.retry` or `request` on the end).